### PR TITLE
feat(util): Add lightweight parseUrl function

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

Simplest wrapper for new WHATWG URL parsing API.
Separate PRs which use it in `datasource/`, `manager/` and `workers/` directories are coming.
More complicated cases (mostly, `auth` field in `platform/` directory) will be handled separatedly.

## Context:

Ref: #7797

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by viceice** [comment]
I've a branch for refactoring url usage too. Maybe i forgot to push it. Will do it tomorrow morning, maybe it helps to speed up this. 🤗

**Comment by viceice** [comment]
This also only works with absolute urls.

**Review by viceice — CHANGES_REQUESTED** [review]
needs deconflicting

pushed my branch: https://github.com/renovatebot/renovate/tree/refactor/url-module

**Comment by rarkins** [comment]
@viceice shall we use yours instead?

**Comment by viceice** [comment]
> @viceice shall we use yours instead?

Sure, but it's not fully done. some tests are failing because of relative urls and i'm not sure hos to solve that properly.
Maybe add a `parseUrlLegacy` function for that case?

File: lib/util/url.ts [hunk 1]
```diff
@@ -48,3 +48,11 @@ export function validateUrl(url?: string, httpOnly = true): boolean {
     return false;
   }
 }
+
+export function parseUrl(url: string): URL | null {
+  try {
+    return new URL(url);
+  } catch (err) /* istanbul ignore next */ {
```
Comment (viceice) [inline lib/util/url.ts:55]:
```suggestion
  } catch (err) {
```

This is already tested 🙃

**Review by viceice — COMMENTED** [review]
We should use this pr and i'll rebase my branch onto this

**Review by viceice — DISMISSED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**